pkg/instrumentors/bpf/google/golang/org/grpc/bpf/probe.bpf.c

```diff
@@ -111,84 +141,78 @@ int uprobe_ClientConn_Invoke_Returns(struct pt_regs *ctx)
 {
     u64 context_pos = 3;
     void *context_ptr = get_argument(ctx, context_pos);
-    void *grpcReq_ptr = bpf_map_lookup_elem(&context_to_grpc_events, &context_ptr);
+    void *key = get_consistent_key(ctx, context_ptr);
+    void *grpcReq_ptr = bpf_map_lookup_elem(&grpc_events, &key);
     struct grpc_request_t grpcReq = {};
     bpf_probe_read(&grpcReq, sizeof(grpcReq), grpcReq_ptr);
 
     grpcReq.end_time = bpf_ktime_get_ns();
     bpf_perf_event_output(ctx, &events, BPF_F_CURRENT_CPU, &grpcReq, sizeof(grpcReq));
-    bpf_map_delete_elem(&context_to_grpc_events, &context_ptr);
-
+    bpf_map_delete_elem(&grpc_events, &key);
+    stop_tracking_span(&grpcReq.sc);
     return 0;
 }
 
-// func (t *http2Client) createHeaderFields(ctx context.Context, callHdr *CallHdr) ([]hpack.HeaderField, error)
-SEC("uprobe/Http2Client_createHeaderFields")
-int uprobe_Http2Client_CreateHeaderFields(struct pt_regs *ctx)
+// func (l *loopyWriter) headerHandler(h *headerFrame) error
```

Comment:
Is it intentional to monitor `loopyWriter_headerHandler` instead of `Http2Client_createHeaderFields`? Do we need to still monitor `http2Client_NewStream` too?

Can you say anymore about why it's better to use this instead?

Reply:
Added description to PR